src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -341,11 +401,40 @@ protected override void VisitSwitchSection(BoundSwitchSection node, bool isLastS
                                         var field = (FieldSymbol)AsMemberOfType(inputType, e.Field);
                                         var type = field.TypeWithAnnotations;
                                         var output = new BoundDagTemp(e.Syntax, type.Type, e);
-                                        int outputSlot = GetOrCreateSlot(field, inputSlot, forceSlotEvenIfEmpty: true);
+                                        int outputSlot = -1;
+                                        var originalTupleElement = e.Input.IsOriginalInput && !originalInputElementSlots.IsDefault
```

Comment:
Should the declaration of `originalInputElementSlots` be moved closer to the usage?

Reply:
oh, I think that its current place is the only way to avoid recomputing it. Disregard this :)

Reply:
Right, we want to compute the input slots once for the whole DAG

---
In reply to: [574006304](https://github.com/dotnet/roslyn/pull/51001#discussion_r574006304) [](ancestors = 574006304)